packages/tc-api-rest-handlers/lib/events/__tests__/docstore-events.spec.js

```diff
@@ -0,0 +1,355 @@
+const AWS = require('aws-sdk');
+const { docstore } = require('@financial-times/tc-api-s3-document-store');
+const events = require('..');
+
+const emitSpy = jest.fn();
+events.emitter.emit = emitSpy;
+
+const { setupMocks } = require('../../../../../test-helpers');
+const { postHandler, patchHandler, absorbHandler } = require('../../..');
+
+const eventTester = desiredAction => (type, code, updatedProperties) => {
+	const matchingEvents = emitSpy.mock.calls.filter(
+		([action, event]) =>
+			action === desiredAction &&
+			event.action === desiredAction &&
+			event.type === type &&
+			event.code === code,
```

Comment:
Nit: we can use `emitSpy.mock.calls.find(...)` instead of filter then we can avoid continuous array indexing at assertion of `expect(matchingEvents[1]).toMatchObject(referenceObject);`

Reply:
I did have that, but that is vulnerable to sending multiple events when only one is intended